Source/ASDisplayNode.mm

```diff
@@ -2898,27 +2898,48 @@ - (void)didExitHierarchy
   // same runloop.  Strategy: strong reference (might be the last!), wait one runloop, and confirm we are still outside the hierarchy (both layer-backed and view-backed).
   // TODO: This approach could be optimized by only performing the dispatch for root elements + recursively apply the interface state change. This would require a closer
   // integration with _ASDisplayLayer to ensure that the superlayer pointer has been cleared by this stage (to check if we are root or not), or a different delegate call.
-  if (ASInterfaceStateIncludesVisible(_pendingInterfaceState)) {
-    void(^exitVisibleInterfaceState)(void) = ^{
-      // This block intentionally retains self.
-      __instanceLock__.lock();
-      unsigned isStillInHierarchy = _flags.isInHierarchy;
-      BOOL isVisible = ASInterfaceStateIncludesVisible(_pendingInterfaceState);
-      ASInterfaceState newState = (_pendingInterfaceState & ~ASInterfaceStateVisible);
-      __instanceLock__.unlock();
-
-      if (!isStillInHierarchy && isVisible) {
-        if (![self supportsRangeManagedInterfaceState]) {
-          newState = ASInterfaceStateNone;
-        }
-        self.interfaceState = newState;
+  if (!ASCATransactionQueue.sharedQueue.enabled) {
+    if (![self supportsRangeManagedInterfaceState]) {
+      self.interfaceState = ASInterfaceStateNone;
+    } else {
+      if (ASInterfaceStateIncludesVisible(self.interfaceState)) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+          // This block intentionally retains self.
+          __instanceLock__.lock();
+          unsigned isInHierarchy = _flags.isInHierarchy;
+          BOOL isVisible = ASInterfaceStateIncludesVisible(_interfaceState);
+          ASInterfaceState newState = (_interfaceState & ~ASInterfaceStateVisible);
+          __instanceLock__.unlock();
+          if (!isInHierarchy && isVisible) {
+            self.interfaceState = newState;
+          }
+        });
       }
-    };
+    }
+  } else {
+    // TODO(wsdwsd0829): rework on this piece code that alter interface behavior of non-interfaceCoalescing.
```

Comment:
// Rework this code to ensure that interface state behavior is not altered when ASCATransactionQueue is disabled.